.github/workflows/cd-workflow.yml

```diff
@@ -7,9 +7,53 @@ on:
   pull_request:
     branches: 
       - main
-
 jobs:
+  code-test:
+# Kicks off the workflow and prepares Github to run coverage test using a ubuntu-latest container.
+    name: Code Test
+    runs-on: ubuntu-latest
+    env:
+      OS: ubuntu-latest
+      PYTHON: '3.9'
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+# Install dependencies required for tests
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r ./requirements.txt
+# Start running test scripts and generate the coverage report
+    - name: Generate Coverage Report
+      env:
+        GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+      run: |
+        pip install pytest
+        pip install pytest-cov
+        echo "Cloning private repository"
+        git clone https://micson-us:${{ secrets.ACCESS_TOKEN }}@github.com/OpenLXP/openlxp-private.git
```

Comment:
What is `micson-us` in this context?  Could it be something more like `openlxp-host`?

Reply:
Hi Matt - that is my username. I will create a service account to run these workflows. openlxp-host is a great name for it!